infrastructure/index.ts

```diff
@@ -341,6 +304,67 @@ function translateRedirect(filePath: string, redirect: string): string {
     throw new Error(`The redirect "${redirect}" in "${filePath}" is not in an expected format.`);
 }
 
+// Find all Hugo-generated meta-refresh redirects and convert them to proper
+// (Pulumi-managed) AWS web-page redirects.
+// https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html
+const redirectPaths = new Set<string>();
+glob.sync(`${webContentsRootPath}/**/*.html`).map(filePath => {
+    const relativeFilePath = filePath.replace(webContentsRootPath + "/", "");
+    const redirect = getMetaRefreshRedirect(filePath);
+
+    if (redirect) {
+        redirectPaths.add(relativeFilePath);
+
+        const redirectObject = new aws.s3.BucketObject(
+            relativeFilePath,
+            {
+                acl: "public-read",
+                key: relativeFilePath,
+                bucket: contentBucket,
+                source: new pulumi.asset.FileAsset("/dev/null"), // Empty file.
+                websiteRedirect: translateRedirect(filePath, redirect),
+            },
+            {
+                parent: contentBucket,
+            },
+        );
+    }
+});
+
+// If the current run is an update (i.e., not a preview), zip up the contents of the
+// website directory and upload the archive to S3.
+if (!pulumi.runtime.isDryRun()) {
+
+    pulumi.all([ archiveBucket.id, contentBucket.id ]).apply(async ([ archiveBucketId, contentBucketId ]) => {
+        const s3 = new aws.sdk.S3();
+
+        // Tar up the files in the `public` directory.
+        const archivePath = tmp.fileSync({ postfix: ".tgz" }).name;
+        tar.c(
+            {
+                gzip: true,
+                sync: true,
+                // Include only files not flagged as redirects.
+                filter: (filePath, stat) => !redirectPaths.has(filePath),
+                file: archivePath,
+                C: config.pathToWebsiteContents,
+                portable: true,
+            },
+            fs.readdirSync(config.pathToWebsiteContents),
+        );
+
+        // Upload the archive.
+        const result = s3.putObject({
```

Comment:
Do we need to `await` the promise to prevent it from being unresolved?

```suggestion
        const result = await s3.putObject({
```